kinesis-video-gstreamer-plugin/src/gstkvssink.cpp

```diff
@@ -1112,15 +1106,16 @@ gst_kvs_sink_handle_buffer (GstCollectPads * pads,
             data->first_pts = buf->pts;
         }
         if (data->producer_start_time == GST_CLOCK_TIME_NONE) {
-            data->producer_start_time = chrono::duration_cast<nanoseconds>(
+            data->producer_start_time = (uint64_t) chrono::duration_cast<nanoseconds>(
                     systemCurrentTime().time_since_epoch()).count();
         }
         buf->pts += data->producer_start_time - data->first_pts;
     }
 
     put_frame(data->kinesis_video_stream, info.data, info.size,
               std::chrono::nanoseconds(buf->pts),
-              std::chrono::nanoseconds(buf->dts), kinesis_video_flags, track_id, data->frame_count);
+              std::chrono::nanoseconds(buf->dts), kinesis_video_flags, track_id, data->frame_count,
+              kvssink->timecode_scale_milliseconds);
```

Comment:
This doesn't change during the run. Set it as part of the info structure as there is no need to pass this as a param.

Also, think if your info could have a back pointer to GstKvsSink parent object which has the necessary info

Reply:
I removed the down scaling. PIC's frame ordering should be able to handle it.